spec/presenters/single_content_item/single_content_item_presenter_spec.rb

```diff
@@ -98,6 +98,104 @@
+
+      expect(subject.total_pviews).to eq('500,000,000')
+    end
+
+    it 'correctly formats number for a small value' do
+      current_period_data[:time_series_metrics] = [{ name: 'pviews', total: 5 }, { name: 'upviews', total: 0 }]
+
+      expect(subject.total_pviews).to eq('5')
+    end
+  end
+
+  describe '#total_searches' do
+    it 'correctly formats number for a value in the millions' do
+      current_period_data[:time_series_metrics] = [{ name: 'searches', total: 500_000_000 }, { name: 'upviews', total: '0' }, { name: 'pviews', total: 0 }]
+
+      expect(subject.total_searches).to eq('500,000,000')
+    end
+
+    it 'correctly formats number for a small value' do
+      current_period_data[:time_series_metrics] = [{ name: 'searches', total: 5 }, { name: 'pviews', total: 5 }, { name: 'upviews', total: 0 }]
+
+      expect(subject.total_searches).to eq('5')
+    end
+  end
+
+  describe '#total_satisfaction' do
+    it 'correctly formats number for a value in the millions' do
+      current_period_data[:time_series_metrics] = [{ name: 'satisfaction', total: 1 }, { name: 'upviews', total: '0' }, { name: 'pviews', total: 0 }]
+
+      expect(subject.total_satisfaction).to eq('100%')
+    end
+
+    it 'correctly formats number for a small value' do
+      current_period_data[:time_series_metrics] = [{ name: 'satisfaction', total: 0.01 }, { name: 'pviews', total: 5 }, { name: 'upviews', total: 0 }]
+
+      expect(subject.total_satisfaction).to eq('1%')
+    end
+  end
+
+  describe '#total_feedex' do
+    it 'correctly formats number for a value in the millions' do
+      current_period_data[:time_series_metrics] = [{ name: 'feedex', total: 1_000 }, { name: 'upviews', total: '0' }, { name: 'pviews', total: 0 }]
+
+      expect(subject.total_feedex).to eq('1,000')
+    end
+
+    it 'correctly formats number for a small value' do
+      current_period_data[:time_series_metrics] = [{ name: 'feedex', total: 1 }, { name: 'pviews', total: 5 }, { name: 'upviews', total: 0 }]
+
+      expect(subject.total_feedex).to eq('1')
+    end
+  end
+
+  describe '#total_words' do
+    it 'correctly formats number for a value in the millions' do
+      current_period_data[:edition_metrics] = [{ name: 'words', value: 5_000 }]
+
+      expect(subject.total_words).to eq('5,000')
+    end
+
+    it 'correctly formats number for a small value' do
+      current_period_data[:edition_metrics] = [{ name: 'words', value: 50 }]
+
+      expect(subject.total_words).to eq('50')
+    end
+  end
+
+  describe '#total_pdf_count' do
+    it 'correctly formats number for a value in the millions' do
+      current_period_data[:edition_metrics] = [{ name: 'pdf_count', value: 5_000 }]
+
+      expect(subject.total_pdf_count).to eq('5,000')
+    end
+
+    it 'correctly formats number for a small value' do
+      current_period_data[:edition_metrics] = [{ name: 'pdf_count', value: 50 }]
+
+      expect(subject.total_pdf_count).to eq('50')
+    end
+  end
```

Comment:
There's quite a lot of code here. Given that it's common functionality for different fields, are there any Rspec features that would allow to express the same tests more concisely?

Reply:
Re: my comment about creating generalised `formatted_metric` function
That would minimise tests here and write simpler tests for `MetricsFormatterHelper:: format_metric_value` which would ensure the correct format.

Reply:
I've taken Sean's [suggestion and removed the repetitive tests](https://github.com/alphagov/content-data-admin/pull/328/commits/169989ee9fcee33ace6daa3fe98a069ad6d0edc6).